### PR TITLE
feat: integrate with funkit for instant deposits

### PR DESCRIPTION
**Pull request by rosepuppy**
https://github.com/user-attachments/assets/8fee662f-5ef6-4400-84dc-c57c940f423d

Integrate with fun-kit for instant deposit

The deposit form for on fun-kit and standard will include a toggle between instant and standard.

Instant deposit will trigger a new notification type `NotificationType.FunkitDeposit`

To test on mainnet, use the mainnet staging link here:
https://v4-staging-git-funkit-integration-mainnet-test-dydxprotocol.vercel.app/trade/ETH-USD

### CODE REVIEW

File: src/constants/funkit.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        showTokenAmount: false,
+        customRecipientConfig: {
+          destinationIcon: <LogoShortIcon id="funkit-dydx-logo" width={15} height={15} />,
+        },
+      },
+    },
+  }) as FunkitConfig;
+
+const ACCENT_COLOR = 'var(--color-accent)';
+const ACCENT_COLOR_WASH = 'rgb(114,122,255)'; // dydx uses brightness
```
Comment (jaredvu) [inline src/constants/funkit.tsx:37]: var(--color-accent-faced) ?

File: src/constants/funkit.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const OFF_BACKGROUND_DARK = 'var(--color-layer-4)';
+const OFF_BACKGROUND_LIGHT = 'var(--color-layer-1)';
```
Comment (jaredvu) [inline src/constants/funkit.tsx:39]: why diff? css var property should change on theme change

File: src/hooks/useFunkitBuyNobleUsdc.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        dispatch(openDialog(DialogTypes.Deposit({ depositType: 'standard' })));
+      },
+    }),
+    [dispatch]
+  );
+  const { beginCheckout } = useFunkitCheckout(config);
+  const startCheckout = useCallback(async () => {
+    setTheme(appThemeSetting === AppTheme.Light ? (lightMode as any) : (darkMode as any));
+    await beginCheckout({
+      modalTitle: 'Deposit',
```
Comment (jaredvu) [inline src/hooks/useFunkitBuyNobleUsdc.ts:48]: stringGetter

File: src/hooks/useFunkitBuyNobleUsdc.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      iconSrc: TOKEN_ICON_SRC,
+      targetChain: TOKEN_CONTRACT_CHAIN_ID,
+      targetAsset: TOKEN_CONTRACT_ADDRESS as `0x${string}`,
```
Comment (jaredvu) [inline src/hooks/useFunkitBuyNobleUsdc.ts:51]: `Address` type?

Comment (jaredvu) [reply src/hooks/useFunkitBuyNobleUsdc.ts:51]: EvmAddress

File: src/hooks/useFunkitBuyNobleUsdc.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      targetAssetAmount: DEFAULT_USDC_AMT,
+      targetAssetTicker: TOKEN_SYMBOL,
+      checkoutItemTitle: TOKEN_SYMBOL,
+      customRecipient: dydxAddress,
+      expirationTimestampMs: CHECKOUT_EXPIRATION_MS,
+    });
+  }, [appThemeSetting, beginCheckout, darkMode, dydxAddress, lightMode, setTheme]);
```
Comment (jaredvu) [inline src/hooks/useFunkitBuyNobleUsdc.ts:58]: check to see if `beginCheckout` is stable

File: src/hooks/useNotificationTypes.tsx [hunk 6]
```diff
@@ -212,6 +216,55 @@ export const notificationTypes: NotificationTypeConfig[] = [
...
+      useEffect(() => {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const deposit of Object.values(funkitDeposits)) {
+          const { checkoutId, status } = deposit;
+          trigger(
+            checkoutId,
+            {
+              icon: <Icon iconName={IconName.FunkitInstant} tw="text-color-accent" />,
+              title:
+                status === 'COMPLETED' || !ErrorStatuses.includes(status ?? '')
```
Comment (jaredvu) [inline src/hooks/useNotificationTypes.tsx:235]: ErrorStatuses can be enum.

File: src/lib/wagmi.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 // Custom connectors
+import { createFunkitWagmiConfig } from '@funkit/connect';
 import type { PrivyClientConfig } from '@privy-io/react-auth';
-import { createConfig } from '@privy-io/wagmi';
```
Comment (jaredvu) [inline src/lib/wagmi.ts:3]: will we lose custom privy logic that they shim into createConfig. Might be good idea to ask privy guys

File: src/hooks/useFunkitBuyNobleUsdc.ts (hunk 5 above)
Comment (HerringtonDarkholme) [reply src/hooks/useFunkitBuyNobleUsdc.ts]: I have checked about our codebase and the beginCheckout is a stable reference unless the input argument of `useFunkitCheckout` is changed.

In this case, `config` is memoized so `beginCheckout` should be stable.

File: src/hooks/useNotificationTypes.tsx (hunk 6 above)
Comment (rosepuppy) [reply src/hooks/useNotificationTypes.tsx:235]: It was a bit challenging to get the types right, I think I can do this in a follow up

**Review by jaredvu — APPROVED** [review]

**Pull request merged**